### PR TITLE
Bugfix fx3808 [v98] Customize home page show from combined menu fix

### PR DESCRIPTION
**Pull request by lmarceau**
- Issue https://mozilla-hub.atlassian.net/browse/FXIOS-3808
- Also rename function name for better clarity on what it does

### CODE REVIEW

File: Client/Extensions/UIViewControllerExtension.swift [hunk 1]
```diff
@@ -70,7 +70,7 @@ extension UIViewController {
         } else {
             themedNavigationController.modalPresentationStyle = .fullScreen
         }
-        present(viewController: themedNavigationController, animated: true)
+        presentDismissIfNeeded(viewController: themedNavigationController, animated: true)
```
Comment (electricRGB) [inline Client/Extensions/UIViewControllerExtension.swift:73]:
```suggestion
        presentWithModalDismissIfNeeded(themedNavigationController, animated: true)
```

File: Client/Extensions/UIViewControllerExtension.swift [hunk 2]
```diff
@@ -79,7 +79,7 @@ extension UIViewController {
 
     /// A convenience function to dismiss modal presentation views if they are
     /// currently presented.
-    func present(viewController: UIViewController, animated: Bool) {
+    func presentDismissIfNeeded(viewController: UIViewController, animated: Bool) {
```
Comment (electricRGB) [inline Client/Extensions/UIViewControllerExtension.swift:82]:
```suggestion
    func presentWithModalDismissIfNeeded(_ : UIViewController, animated: Bool) {
```

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -1351,7 +1351,7 @@ class BrowserViewController: UIViewController {
             popoverPresentationController.delegate = self
         }
 
-        present(viewController: controller, animated: true)
+        presentDismissIfNeeded(viewController: controller, animated: true)
```
Comment (electricRGB) [inline Client/Frontend/Browser/BrowserViewController.swift:1354]:
```suggestion
        presentWithModalDismissIfNeeded(controller, animated: true)
```

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 4]
```diff
@@ -1454,7 +1454,7 @@ class BrowserViewController: UIViewController {
 
         let controller = ThemedNavigationController(rootViewController: settingsTableViewController)
         controller.presentingModalViewControllerDelegate = self
-        self.present(controller, animated: true, completion: nil)
+        presentDismissIfNeeded(viewController: controller, animated: true)
```
Comment (electricRGB) [inline Client/Frontend/Browser/BrowserViewController.swift:1457]:
```suggestion
        presentWithModalDismissIfNeeded(controller, animated: true)
```

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift [hunk 5]
```diff
@@ -197,7 +197,7 @@ extension BrowserViewController: ToolBarActionMenuDelegate {
     }
 
     func showViewController(viewController: UIViewController) {
-        present(viewController: viewController, animated: true)
+        presentDismissIfNeeded(viewController: viewController, animated: true)
```
Comment (electricRGB) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift:200]:
```suggestion
        presentWithModalDismissIfNeeded(viewController, animated: true)
```

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 6]
```diff
@@ -1123,7 +1123,7 @@ extension FirefoxHomeViewController: HomePanelContextMenu {
                 popoverController.permittedArrowDirections = [.up, .down, .left]
                 popoverController.delegate = self
             }
-            self.present(viewController: controller, animated: true)
+            self.presentDismissIfNeeded(viewController: controller, animated: true)
```
Comment (electricRGB) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:1126]:
```suggestion
            self.presentWithModalDismissIfNeeded(controller, animated: true)
```

**Review by electricRGB — COMMENTED** [review]
My suggestion.

**Review by yoanarios — APPROVED** [review]
Tested and is working 🚀

**Pull request merged**